src/reducers/greetings.js

```diff
@@ -0,0 +1,24 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const API_URL = 'http://localhost:3000/greetings/random';
+const GET_INFO = 'hello_rails_react/greetings/GET_INFO';
+const initialState = {
+  greetings: '',
+};
+
+const greetingReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case `${GET_INFO}/fulfilled`:
+      return { ...state, greetings: action.payload };
+    default:
+      return state;
+  }
+};
+
+export const fetchGreeting = createAsyncThunk(GET_INFO, async () => {
+  const response = await fetch(API_URL);
+  const result = await response.json();
+  return result;
+});
+
+export default greetingReducer;
```

Comment:
- [ ] Dear @IngDavidTM, I must admit that you have done a great job implementing the functionalities. But as of now, your app has one flaw left to be addressed. which is called `CORS policy`. Because of this, I could not fetch the data from the back end. This happens when the server only replays to clients who have access to use it. [Read more here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors) or check this block on [how to fix CORS issue on rails](https://www.stackhawk.com/blog/rails-cors-guide/)

https://user-images.githubusercontent.com/69077061/221239343-11c43155-0781-45e5-b329-12fc25651492.mp4